test/dep-graph.spec.ts

```diff
@@ -106,48 +106,76 @@ class TestFixture {
     expect(baselinePathsToRoot).toEqual(withRuntimeDepsPathsToRoot);
   });
 
-  it('does not crash when parsing a project with no dependencies', async () => {
-    // Generate some random project
-    const files: types.DotNetFile[] = [
-      {
+  it.each([
+    {
+      description: 'no dependencies',
+      files: [
+        {
+          name: 'no_deps.csproj',
+          contents: `
+<Project Sdk='Microsoft.NET.Sdk'>
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+  </PropertyGroup>
+</Project>
+`,
+        },
+      ],
+    },
+    {
+      description: 'preserved compilation context',
+      files: [
+        {
+          name: 'withCompilationContext.csproj',
+          contents: `
+<Project Sdk='Microsoft.NET.Sdk.Web'>
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <PreserveCompilationContext>true</PreserveCompilationContext>
+  </PropertyGroup>
+</Project>
+`,
+        },
+      ],
+    },
+  ])(
+    'succeeds to generate a dependency graph for a project with: $description',
+    async ({ description, files }) => {
+      const fixtureName = description.replace(/\s/g, '-');
+      const program = {
         name: 'program.cs',
         contents: `
 using System;
+
 class TestFixture {
     static public void Main(String[] args)
     {
-      var client = new System.Net.Http.HttpClient();
       Console.WriteLine("Hello, World!");
     }
 }
 `,
-      },
-      {
-        name: 'no_deps.csproj',
-        contents: `
-<Project Sdk='Microsoft.NET.Sdk'>
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
-  </PropertyGroup>
-</Project>
-`,
-      },
-    ];
-    const fixtureName = 'noDeps';
-    projectDirs[fixtureName] = codeGenerator.generate('fixtures', files);
-    const tempDir = projectDirs[fixtureName];
-    await dotnet.restore(tempDir);
+      };
 
-    const results = await nugetParser.buildDepGraphFromFiles(
-      tempDir,
-      'obj/project.assets.json',
-      ManifestType.DOTNET_CORE,
-      false,
-    );
-    expect(results.length).toEqual(1);
-    expect(results[0].dependencyGraph).toBeDefined();
-  });
+      const filesWithProgram = files.concat(program);
+
+      projectDirs[fixtureName] = codeGenerator.generate(
+        'fixtures',
+        filesWithProgram,
+      );
+      const tempDir = projectDirs[fixtureName];
+      await dotnet.restore(tempDir);
+
+      const results = await nugetParser.buildDepGraphFromFiles(
+        tempDir,
+        'obj/project.assets.json',
+        ManifestType.DOTNET_CORE,
+        false,
+      );
+      expect(results.length).toEqual(1);
+      expect(results[0].dependencyGraph).toBeDefined();
```

Comment:
These tests were all project constellations that caused the plugin to crash before any depGraph was generated. And we're still getting errors in like that. So just the fact that we have a depGraph is enough to assert. We already have thorough test coverage of the actual dependency graph generation other places in the tests.